 ## This interactive python is to analyse and test desciprive anlysis before adding the usefula codes in the dashboard

In [16]:
import pandas as pd
import plotly.express as px

In [17]:
#Reading the selected dataset
df = pd.read_csv("processed_data.csv")
df.head()

Unnamed: 0     Sex   Age  Hypertension  Heart Disease  Married  \
0           0    Male  67.0         False           True     True   
1           1  Female  61.0         False          False     True   
2           2    Male  80.0         False           True     True   
3           3  Female  49.0         False          False     True   
4           4  Female  79.0          True          False     True   

       Work Type Residence Type  Glucose   BMI         Smoking?  Stroke  
0        Private          Urban   228.69  36.6  formerly smoked    True  
1  Self-employed          Rural   202.21   0.0     never smoked    True  
2        Private          Rural   105.92  32.5     never smoked    True  
3        Private          Urban   171.23  34.4           smokes    True  
4  Self-employed          Rural   174.12  24.0     never smoked    True

In [18]:
total_patients = len(df)
stroke_rate = (df['Stroke'] == 1).mean() * 100
mean_age = df['Age'].mean()
heart_disease_rate = (df['Heart Disease'] == 1).mean() * 100

print(f"Total patients: {total_patients}\nStroke rate: {stroke_rate}\nMean age: {mean_age}\nHeart disease rate: {heart_disease_rate}")

Total patients: 5110
Stroke rate: 4.87279843444227
Mean age: 43.226614481409
Heart disease rate: 5.401174168297456


In [19]:
#Gender analysis
#What is the gender distribution in the Stroke prediction dataset?

gender_counts = df['Sex'].value_counts(dropna=False)

# Create Plotly pie chart
fig_gender = px.pie(
    values=gender_counts.values,
    names=gender_counts.index,
    title='Gender Distribution',
    color_discrete_map={'Male': '#1f77b4', 'Female': '#ff7f0e'}
)

# Adjust layout
fig_gender.update_layout(height=400)

# Display chart
fig_gender.show()


### Reflection
The pie chart that illustrated the distribution of participants: males comprised 58.6%, females accounted for 41.4%, and others represented 0.01%.

In [20]:
#Age Distribution by stroke
#What is the median age of the patients with stroke and without stroke?

fig_box = px.box(
    df,
    y='Age',
    x='Stroke',
    title='Age Distribution: Stroke vs No Stroke',
    labels={'Age': 'Age (years)', 'Stroke': 'Had Stroke'},
    color='Stroke',
    color_discrete_map={0: '#3498db', 1: '#e74c3c'}
)

# Customize axes and layout
fig_box.update_xaxes(ticktext=['No Stroke', 'Stroke'], tickvals=[0, 1])
fig_box.update_layout(height=400, showlegend=False)

# Display chart
fig_box.show()

### Reflection
This chart indicated that the median age for stroke patients is lower, at 40, with the majority of individuals falling within the 20 to 60 age range. In contrast, the median age of non-stroke patients is 65, with a narrower age distribution

In [21]:
#Heart Diseases and Storke
grp = df.groupby(['Heart Disease', 'Stroke']).size().reset_index(name='Count')

# % within each Heart Disease group (row-wise)
grp['Percentage within Heart Disease (%)'] = (
    grp['Count'] / grp.groupby('Heart Disease')['Count'].transform('sum') * 100
).round(2)

grp

Heart Disease  Stroke  Count  Percentage within Heart Disease (%)
0          False   False   4632                                95.82
1          False    True    202                                 4.18
2           True   False    229                                82.97
3           True    True     47                                17.03

### Reflection
The pivot table shows that stoke rate in patient with heart disease is significantly higer than that of no heart disease patients with 17.03% and 4.18 % respectively.

In [22]:
#Smoking and Storke
grp_smoke = df.groupby('Smoking?')['Stroke'].mean().reset_index()

# Convert stroke rate to percentage
grp_smoke['Stroke Rate (%)'] = grp_smoke['Stroke'] * 100

# Create bar chart
fig_smoke = px.bar(
    grp_smoke,
    x='Smoking?',
    y='Stroke Rate (%)',
    title='Stroke Rate by Smoking Status',
    labels={'Smoking?': 'Smoking Status', 'Stroke Rate (%)': 'Stroke Rate (%)'},
    color_discrete_sequence=["#43a6df"]
)

# Adjust layout
fig_smoke.update_layout(height=400)

# Display chart
fig_smoke.show()

### Reflection
The graph shows that the rates are higher among former and current smokers, with figures of 7.9% and 5.3% respectively, compared to never smokers, who have a rate of under 4.7%

In [23]:
#Work type, residence, heart disease

grp_work = df.groupby(['Work Type', 'Residence Type'])['Heart Disease'].mean().reset_index()
grp_work['Heart Disease Rate (%)'] = grp_work['Heart Disease'] * 100
grp_work

Work Type Residence Type  Heart Disease  Heart Disease Rate (%)
0       Govt_job          Rural       0.051282                5.128205
1       Govt_job          Urban       0.057971                5.797101
2   Never_worked          Rural       0.000000                0.000000
3   Never_worked          Urban       0.000000                0.000000
4        Private          Rural       0.055404                5.540356
5        Private          Urban       0.052632                5.263158
6  Self-employed          Rural       0.094148                9.414758
7  Self-employed          Urban       0.103286               10.328638
8       children          Rural       0.000000                0.000000
9       children          Urban       0.002882                0.288184

### Reflection
The table shows that the highest proportion (10%) among self-employed individuals living in urban areas, while the lowest proportion (5.6%) was among those with private jobs in urban areas

In [24]:
#Average glucose level in Storke/Non-Stroke patients
# Group by stroke status and calculate mean glucose level
grp_glucose = df.groupby('Stroke')['Glucose'].mean().reset_index()

# Create interactive bar chart
fig_glucose = px.bar(
    grp_glucose,
    x='Stroke',
    y='Glucose',
    title='Average Glucose Level by Stroke Status',
    labels={'Stroke': 'Stroke Status', 'Glucose': 'Average Glucose Level (mg/dL)'},
    color_discrete_sequence=["#250feb"]
)

# Customize x-axis and layout
fig_glucose.update_xaxes(
    ticktext=['No Stroke', 'Stroke'],
    tickvals=[0, 1]
)
fig_glucose.update_layout(height=400)

# Display
fig_glucose.show()

### Reflection
The graph indicated that stroke patients had a significantly higher average glucose level of 132 mg/dL compared to 104 mg/dL in non-stroke patients

In [25]:
# Average BMI in Storke/Non-Stroke patients
grp_bmi = df.groupby('Hypertension')['BMI'].mean()
grp_bmi

Hypertension
False    27.523287
True     29.918675
Name: BMI, dtype: float64

### Reflection
The result shows that the mean BMI values were relatively close, with stroke patients averaging 29.9 and non-stroke patients averaging 27.5